src/gpu/generic/sycl/ref_deconvolution.hpp

```diff
@@ -0,0 +1,155 @@
+            using sm = primitive_attr_t::skip_mask_t;
+
+            const memory_desc_wrapper data_d(src_md());
+            const memory_desc_wrapper diff_weights_d(diff_weights_md());
+            const memory_desc_wrapper diff_dst_d(diff_dst_md());
+
+            const bool ok = desc()->prop_kind == prop_kind::backward_weights
+                    && check_convolution_work_amount(diff_weights_d, OC())
+                    && md_dims_in_range(src_md())
+                    && set_default_formats()
+                    && check_convolution_data_types(
+                            data_d, diff_weights_d, diff_dst_d)
+                    && check_convolution_formats(
+                            data_d, diff_weights_d, diff_dst_d)
+                    && attr()->has_default_values(sm::scales_runtime
+                            | sm::zero_points_runtime | sm::post_ops
+                            | sm::sum_dt)
+                    && IMPLICATION(!attr()->scales_.has_default_values(),
+                            check_scales_mask())
+                    && post_ops_ok();
+            if (!ok) return status::unimplemented;
+
+            return init_conf();
+        }
+
+        sycl_convolution_conf_t conf_;
+
+    private:
+        status_t init_conf();
+
+        bool set_default_formats_common_template(memory_desc_t &src_md,
+                format_tag_t src_tag, memory_desc_t &wei_md,
+                format_tag_t wei_tag, memory_desc_t &dst_md,
+                format_tag_t dst_tag, memory_desc_t &bia_md) {
+            using namespace format_tag;
+
+#define IS_OK(f) \
+    do { \
+        if ((f) != status::success) return false; \
+    } while (0)
+    
+            if (src_md.format_kind == format_kind::any
+                    && !utils::one_of(src_tag, any, undef))
+                IS_OK(memory_desc_init_by_tag(src_md, src_tag));
+            if (dst_md.format_kind == format_kind::any
+                    && !utils::one_of(dst_tag, any, undef))
+                IS_OK(memory_desc_init_by_tag(dst_md, dst_tag));
+            if (wei_md.format_kind == format_kind::any
+                    && !utils::one_of(wei_tag, any, undef))
+                IS_OK(memory_desc_init_by_tag(wei_md, wei_tag));
+            if (with_bias() && bia_md.format_kind == format_kind::any)
+                IS_OK(memory_desc_init_by_tag(bia_md, x));
+#undef IS_OK
+
+            return true;
+        }
+
+        bool set_default_formats() {
+            using namespace format_tag;
+            auto dat_tag = utils::pick(ndims() - 3, nwc, nhwc, ndhwc);
+            auto wei_tag = with_groups()
+                    ? utils::pick(ndims() - 3, goiw, goihw, goidhw)
+                    : utils::pick(ndims() - 3, oiw, oihw, oidhw);
+            return set_default_formats_common_template(src_md_, dat_tag,
+                    diff_weights_md_, wei_tag, diff_dst_md_, dat_tag,
+                    diff_bias_md_);
+        }
+
+        bool check_scales_mask() const {
```

Comment:
In general, scales are used for inference.
Hence scales are typically not supported for bwd by weights.